R/Algorithms.R

```diff
@@ -1,22 +1,109 @@
-#' A random_search implementation
+#' Random Search
 #'
-#' This implements a bare-bones random search algorithm.
-#' This serves as an example of how to implement an optimization algorithm
-#' for use with the IOHexperimenter.
+#' Random walk in \eqn{{0, 1}^d} space  
 #'
 #'
 #' @param IOHproblem An IOHproblem object
 #'
 #' @export
-random_search <- function(IOHproblem) {
-  iter <- 0
-  max_iter <- IOHproblem$dimension
-  while ( iter < max_iter && !IOHproblem$target_hit() ){
-    candidate <- sample(c(0, 1), IOHproblem$dimension, TRUE)
-    IOHproblem$store_parameters(0)
-    IOHproblem$obj_func(candidate)
-    iter <- iter+1
+random_search <- function(dim, obj_func, budget = NULL) {
```

Comment:
I think the algorithms should still just accept a single parameter: the IOHproblem, which contains all the necessary properties such as dimension as propeties; as well as for example the target_hit functions.
This way, there is a simple standard to use, and when you don't want to use a property, you simply ignore it, while otherwise the benchmarking would need to figure out which parameters are required for which problems.